src/ast/mod.rs

```diff
@@ -147,15 +150,121 @@ impl fmt::Display for Ident {
 }
 
 /// A name of a table, view, custom type, etc., possibly multi-part, i.e. db.schema.obj
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+///
+/// # Semantic Equality
+///
+/// This implements equals and hash in terms of the ident *values*, not
+/// including quotes.
+///
+/// ```
+/// use sqlparser::ast::{Ident, ObjectName};
+///
+/// let with = Ident::with_quote('"', "one");
+/// let without = Ident::new("one");
+/// assert_eq!(with.to_string(), "\"one\"");
+/// assert_eq!(without.to_string(), "one");
+/// assert_eq!(ObjectName(vec![with]), ObjectName(vec![without]));
+/// ```
+///
+/// If you need to compare based on quotes see the
+/// [`ObjectName::quoted_string`] method.
+#[derive(Debug, Clone, Eq)]
 pub struct ObjectName(pub Vec<Ident>);
 
+impl PartialEq for ObjectName {
+    fn eq(&self, rhs: &ObjectName) -> bool {
+        if self.0.len() != rhs.0.len() {
+            return false;
+        }
+
+        for (l, r) in self.0.iter().zip(rhs.0.iter()) {
+            if l.value != r.value {
+                return false;
+            }
+        }
+
+        true
+    }
+}
+
+impl Hash for ObjectName {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        for ident in &self.0 {
+            state.write(ident.value.as_bytes());
+        }
+    }
+}
+
 impl fmt::Display for ObjectName {
+    /// Construct an object name including quotes around all internal idents
+    ///
+    /// Uses the format of [`ObjectName::quoted_string`].
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         write!(f, "{}", display_separated(&self.0, "."))
```

Comment:
I think this should use `self.quoted_display()` to avoid skew?

Reply:
oh I thought i did, yeah definitely.